sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasing.scala

```diff
@@ -276,6 +276,16 @@ object GeneratorNestedColumnAliasing {
           val pushedThrough = Project(NestedColumnAliasing
             .getNewProjectList(projectList, nestedFieldsNotOnGenerator), newChild)
 
+          // If the generator output is `ArrayType`, we cannot push through the extractor.
+          // It is because we don't allow field extractor on two-level array,
+          // i.e., attr.field when attr is a ArrayType(ArrayType(...)).
+          // Similarily, we also cannot push through if the child of generator is `MapType`.
+          g.generator.children.head.dataType match {
```

Comment:
Got it. Let me play more with this PR for a while. It seems I need more tests.

Reply:
Thank you!